lib/pbench/agent/results.py

```diff
@@ -199,6 +178,33 @@ def make_result_tb(self, single_threaded: bool = False) -> TarballRecord:
         with mdlog_name.open("w") as fp:
             mdlog.write(fp)
 
+    def make_result_tb(self, single_threaded: bool = False) -> TarballRecord:
+        """Make the result tar ball from result directory.
+
+        The metadata.log file in the result directory is double checked to be
+        sure it is valid, and then the "run.raw_size" and the
+        "pbench.tar-ball-creation-timestamp" fields are added.
+
+        The tar ball is created, verified, and the MD5 sum value is generated.
+
+        Returns a named tuple consisting of the Path object of the created tar
+        ball, its length, and its MD5 checksum value.
+
+        Raises
+          - FileNotFoundError  if the result directory does not have a
+                               metadata.log file
+          - BadMDLogFormat     if the metadata.log file has a pbench.name field
+                               value which does not match the result directory
+                               name.
+          - RuntimeError       if any problems are encountered while creating
+                               or verifying the created tar ball
+        """
+        pbench_run_name = self.result_dir.name
+        try:
+            self.verify_metadata(pbench_run_name)
+        except Exception:
+            raise
```

Comment:
Here's another one which I previously missed:  it doesn't look like this `try` block has any effect!  It catches basically all exceptions and immediately reraises them, which is exactly what would happen if it weren't there.

You should probably remove it.